src/caching/github-actions-cache.ts

```diff
@@ -193,27 +206,41 @@ export class GitHubActionsCache implements Cache {
 
     const version = this.#computeVersion(fingerprint);
     const key = this.#computeCacheKey(script);
-    const url = new URL('_apis/artifactcache/cache', this.#baseUrl);
-    url.searchParams.set('keys', key);
-    url.searchParams.set('version', version);
-
-    using requestResult = this.#request(url);
+    const url = new URL(
+      // See https://github.com/actions/toolkit/blob/930c89072712a3aac52d74b23338f00bb0cfcb24/packages/cache/src/generated/results/api/v1/cache.twirp-client.ts#L91
+      'twirp/github.actions.results.api.v1.CacheService/GetCacheEntryDownloadURL',
```

Comment:
Could this be an absolute path?

Reply:
In the V1 API, the base URL we were passed came with a path component, so it was important not to clobber that by constructing an API URL with an absolute path. The V2 base URL is just `https://results-receiver.actions.githubusercontent.com/`, but I think if it were to change to come with a path, I would want to preserve it.